app/models/digest_run.rb

```diff
@@ -10,17 +10,11 @@ class DigestRun < ApplicationRecord
 
   enum range: { daily: 0, weekly: 1 }
 
-  def mark_complete!
-    completed_at = digest_run_subscribers.maximum(:completed_at) || Time.zone.now
-    update!(completed_at: completed_at)
-  end
-
-  def check_and_mark_complete!
-    mark_complete! unless has_incomplete_subscribers?
-  end
+  def check_if_completed
```

Comment:
Could this be renamed to `check_and_mark_complete`? I wouldn't expect a method called `check_if_completed` to actually update anything

Reply:
Sure, I didn't think `check_and_mark_complete?` seemed right since this didn't return a boolean. I also found `check_and_mark_complete` didn't sound great as it didn't capture the conditional nature if the mark complete. I understand the concern with `check_if_completed` it does sound more like a query than a state change.

I've addressed this by moving the conditional to the worker for this.